__tests__/app.test.js

```diff
@@ -197,4 +197,95 @@ describe('GET /api/articles/', () => {
+                article_id: 7
+            })
+            expect(body).toHaveProperty('created_at');
+        });
+    });
+    test('POST 400: should respond with a 400 if the post body object does not have correct username and/or body properties with values of the correct datatype', () => {
+        const postBody = {
+            faveFood: "peanut butter",
+            body: "The greatest glory in living lies not in never falling, but in rising every time we fall"
+        }
+        return request(app)
+        .post('/api/articles/7/comments')
+        .expect(400)
+        .send(postBody)
+        .then(({body}) => {
+            expect(body.msg).toBe('Invalid input')
+        });
+    });
+    test('POST 400: should respond with a 400 if the post body object has username and body properties but the values are of an incorrect datatype', () => {
+        const postBody = {
+            username: 3,
+            body: ["The greatest glory in living lies not in never falling, but in rising every time we fall"]
+        }
+        return request(app)
+        .post('/api/articles/7/comments')
+        .expect(400)
+        .send(postBody)
+        .then(({body}) => {
+            expect(body.msg).toBe('Invalid input')
+        });
+    });
+    test('POST 400: should respond with a 400 error if given an invalid article_id parameter in the endpoint', () => {
+        const postBody = {
+            username: "lurker",
+            body: "The greatest glory in living lies not in never falling, but in rising every time we fall"
+        }
+        return request(app)
+        .post('/api/articles/forklift/comments')
+        .expect(400)
+        .send(postBody)
+        .then(({body}) => {
+            expect(body.msg).toBe('Invalid input');
+        });
+    });
+    test('POST 404: should respond with a 404 if a valid `article_id` is given in the endpoint, but the article_id does not exist in the database', () => {
+        const postBody = {
+            username: "lurker",
+            body: "The greatest glory in living lies not in never falling, but in rising every time we fall"
+        }
+        return request(app)
+        .post('/api/articles/99999/comments')
+        .expect(404)
+        .send(postBody)
+        .then(({body}) => {
+            expect(body.msg).toBe('No article found for article_id: 99999');
+        });
+    });
+    test('POST 404: should respond with a 404 if a valid username is given, but the corresponding user does not exist in the database', () => {
+        const postBody = {
+            username: "nrmandela",
+            body: "The greatest glory in living lies not in never falling, but in rising every time we fall"
+        }
+        return request(app)
+        .post('/api/articles/7/comments')
+        .expect(404)
+        .send(postBody)
+        .then(({body}) => {
+            expect(body.msg).toBe('User nrmandela not found');
+        });
+    });
+});
+
```

Comment:
So close! I would like one more test: status 201 - ignores any unnecessary properties provided on the post body

Reply:
Gotcha! on it now :)